### PR TITLE
add keyboard suika27melo

### PR DESCRIPTION
**Pull request by suikagiken**
## Description

Add keyboard suikagiken/suika27melo.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* n/a

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/suikagiken/suika27melo/keyboard.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    },
+    "url": "https://github.com/suikagiken/suika27melo",
+    "usb": {
+        "device_version": "1.1.0",
+        "vid": "0x4B48",
+        "pid": "0x0201"
+    },
+    "dynamic_keymap": {
+        "layer_count": 4
+    },
```
Comment (waffle87) [inline keyboards/suikagiken/suika27melo/keyboard.json:29]: Default value.
```suggestion
```

File: keyboards/suikagiken/suika27melo/keyboard.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    "maintainer": "suikagiken",
+    "processor": "atmega32u4",
+    "bootloader": "atmel-dfu",
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "midi":true,
```
Comment (waffle87) [inline keyboards/suikagiken/suika27melo/keyboard.json:13]: Nit pick.
```suggestion
        "midi": true,
```

File: keyboards/suikagiken/suika27melo/readme.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+[こちら](https://github.com/suikagiken/suika27melo/blob/main/buildguide_1.1.md)からご覧下さい。
+
+---
+
+## Overview
+
+suika27melo is a 2-octave piano shaped macro keypad.
+You can use it as a normal macro keypad, or use it as a MIDI keyboard for DAW software input and musical performance.
+It supports VIA/REMAP for changing key assignments, so it can be easily customized after assembly.
```
Comment (waffle87) [inline keyboards/suikagiken/suika27melo/readme.md:25]:
```suggestion
```

File: keyboards/suikagiken/suika27melo/readme.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
+# suika27melo
+(English Follows Japanese)
+
+## 概要
+![suika27melo1](https://github.com/suikagiken/suika27melo/blob/main/img/suika27melo_1.jpg)
+![suika27melo2](https://github.com/suikagiken/suika27melo/blob/main/img/suika27melo_2.jpg)
```
Comment (waffle87) [inline keyboards/suikagiken/suika27melo/readme.md:6]: Could you please rehost these on Imgur.

File: keyboards/suikagiken/suika27melo/readme.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+## Bootloader
+
+Enter the bootloader in 3 ways:
+
+* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (the top left key) and plug in the keyboard
+* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
+* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available
+
+
```
Comment (waffle87) [inline keyboards/suikagiken/suika27melo/readme.md:48]:
```suggestion
```

**Review by suikagiken — COMMENTED** [review]
@waffle87  Thank you for your suggestion.
I have made the corrections.

**Review by waffle87 — APPROVED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/suikagiken/suika27melo/keymaps/default/keymap.json [hunk 6]
```diff
@@ -0,0 +1,12 @@
+{
+  "author": "@suikagiken",
+  "keyboard": "suikagiken/suika85ergo",
```
Comment (fauxpark) [inline keyboards/suikagiken/suika27melo/keymaps/default/keymap.json:3]: This value is wrong; the board does not compile properly in the Configurator. Please fix this.

File: keyboards/suikagiken/suika27melo/keymaps/midi/keymap.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
+{
+  "author": "@suikagiken",
+  "keyboard": "suikagiken/suika85ergo",
```
Comment (fauxpark) [inline keyboards/suikagiken/suika27melo/keymaps/midi/keymap.json:3]: As above.

**Comment by suikagiken** [comment]
@fauxpark Thank you for the advice. I have issued a PR for the corrections here.
https://github.com/qmk/qmk_firmware/pull/24677/

**Pull request merged**